src/main.rs

```diff
@@ -133,12 +138,10 @@ fn print_seq(s: &str) {
 
     // get the translation of this sequence
     let mut peptide = String::new();
-    let n_codons = s.len() / 3;
-    for i in 0..n_codons {
-        let codon = &s[i * 3..(i * 3) + 3]; // take a 3-base slice of the sequence
-        let aa = translate(&codon, i);
+    for (i, codon) in s.as_bytes().windows(3).enumerate() {
+        let aa = translate(codon, i)?;
         // translate and add to the string
-        peptide.push_str(&three_letter_code(aa));
+        peptide.push_str(&three_letter_code(aa)?);
     }
```

Comment:
These are functionally doing the same thing, but I REALLY LOVE the `.windows` method because it makes so, so easy to work with `k`-mers in Rust =]

I'm calling `.as_bytes` here to convert `&str` to its `&[u8]` equivalent without any memory copies.